### DISCUSSION TITLE
Automate the retrieve job without need of the .mcdev-auth.json

### DISCUSSION THREAD

**Discussion by gdibernardo8** [Accenture/sfmc-devtools#349] [General]
Hi, first of all good job for the great tool, this is super useful!

I managed to automate the execution of "mcdev retrieve"  in Azure DevOps, the only thing is that the file .mcdev-auth.json needs to be in the repo and I don't like to keep file with credentials there. I'd like to store it in a more secure way.
Do you have any recommendations for that?

**Comment by gdibernardo8** [comment]
Oh, nevermind I might have found it, testing now :)
mcdev init --y.credentialsName "yourCustomCredentialName" --y.clientId "yourClientIdHere" --y.clientSecret "yourClientSecretHere" --y.tenant "yourTenantSubdomainHere" --y.gitRemoteUrl "https://my.git.server.com/myrepo.git"

**Reply by JoernBerkefeld** [reply]
thats it exatly. though my one devops colleague likes to instead create the .mcdev-auth.json manually in the directory instead. up to you really.
